iot/api/CloudIotSample/Program.cs

```diff
@@ -1376,6 +1491,209 @@ public static object ClearRegistry(string projectId, string cloudRegion, string
         }
         //[END iot_clear_registry]
 
+
+        //[START iot_attach_device]
+        public static object AttachDevice(MqttClient client, string deviceId)
+        {
+            var attachTopic = $"/devices/{deviceId}/attach";
+            Console.WriteLine("Attaching: {0}", attachTopic);
+            var attachPayLoad = "{}";
+            var BinaryData = Encoding.UTF8.GetBytes(attachPayLoad);
+            client.Publish(attachTopic, BinaryData, MqttMsgBase.QOS_LEVEL_AT_LEAST_ONCE, true);
+
+            Console.WriteLine("Waiting for device to attach.");
+            System.Threading.Thread.Sleep(2000);
```

Comment:
Given the call to Sleep(), I would expect a later statement to confirm the the device attached.  Why sleep here?

Reply:
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/1980f31828104d9c39ed4ee11239b1a95ce1f00c/iot/api-client/beta-features/gateway/gateway.py#L463-L465

Reply:
https://github.com/GoogleCloudPlatform/python-docs-samples/blob/1980f31828104d9c39ed4ee11239b1a95ce1f00c/iot/api-client/beta-features/gateway/gateway.py#L432-L442

this is original code which I tried to convert.

Do u think placing Sleep() right after AttachDevice() method will be better?